### PR TITLE
Weekly Sync Dec 10, 2019

### PR DESCRIPTION
**Pull request by daviddias**

### CODE REVIEW

File: _meeting-notes/2019-12-10.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+          - Issue: Dialing in a ring then bootstrapping will still dial everyone (because the DHT is a ring).
+          - Solution: Don't dial everyone.
+          - Solution: Use bootstrap peers to avoid connecting to every peer on bootstrap.
+          - Solution: [TODO] Stagger the bootstrap step so peers bootstrap into an existing DHT instead of a fresh DHT.
+- ==@jimpick==
+    - Test Plan 4
+        - Discussed ideas with David
+        - David "unpacked" it https://github.com/ipfs/testground/issues/96#issuecomment-562877707
+        - Working on further "unpacking" - a new "docs" directly under the testplan with a complete "brain-dump" / plan - work-in-progress, will ask for review on PR tomorrow https://github.com/ipfs/testground/pull/232
+    - Test SideCar and demonstrate how to use it (docs or recorded demo)
```
Comment (daviddias) [inline _meeting-notes/2019-12-10.md]: @jimpick where can I find this docs or demo?

File: _meeting-notes/2019-12-10.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        - I reviewed the Sidecar code, but didn't write a new test yet that uses it
+    - Tested Anton's Infra PR
+- ==@nonsense==
+    - Got the `daemon` PR merged
+    - Got the initial `infra` PR (`terraform` and `ansible` playbooks) reviewed and merged
+    - Took over infra from Jim
+    - Added redirect from http://testground.ipfs.team to github.com/ipfs/testground :) testground.ipfs.team is now managed within the Testground team
+- ==@hacdias==
+    - Added Dependency Tree tracking ([PR](https://github.com/ipfs/testground/pull/204))
```
Comment (daviddias) [inline _meeting-notes/2019-12-10.md]: 👏🏽

File: _meeting-notes/2019-12-10.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    - Added initial tests for Chewing Large Datasets plan ([PR](https://github.com/ipfs/testground/pull/58))
+    - Added daemon support for large datasets plan, as well as two full tests ([PR](https://github.com/ipfs/testground/pull/223))
```
Comment (daviddias) [inline _meeting-notes/2019-12-10.md]: 👏🏽 Nice!

File: _meeting-notes/2019-12-10.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      - @nonsense - this seems to be the lowest priority for the week, right?
+- ==@hacdias==
+  - Keep working on Chewing Strategies (https://github.com/ipfs/testground/issues/93)
+      - Create an issue with the APIs that are missing from Core API and Go-ipfs-api
+      - Should go-ipfs-api and core api have the same interface?
+  - DHT refactory thingy (see @stebalien's) update bellow
+- ==@jimpick==
+  - Finish "brain dump" for Nodes Connectivity planning https://github.com/ipfs/testground/pull/232
+  - Basic libp2p connectivity test + NAT (needs to be broken into sub-tasks)
+  - Review current open bugs and tackle a set of them
```
Comment (daviddias) [inline _meeting-notes/2019-12-10.md]: @jimpick here are some suggestions for you to take on this week:

- https://github.com/ipfs/testground/issues/191
- https://github.com/ipfs/testground/issues/173
- https://github.com/ipfs/testground/issues/122
- https://github.com/ipfs/testground/issues/34
- https://github.com/ipfs/testground/issues/54
- https://github.com/ipfs/testground/issues/118
- https://github.com/ipfs/testground/issues/124
- https://github.com/ipfs/testground/issues/131

Comment (daviddias) [reply _meeting-notes/2019-12-10.md]: @jimpick from this list, please highlight the ones you expect to deliver this week.

File: _meeting-notes/2019-12-10.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+  - [Testing Infra Design](https://github.com/ipfs/testground/blob/master/docs/SPEC.md)
+  - [Testing Inventory](https://github.com/ipfs/testground/blob/master/docs/test-inventory.md)
+  - [Testing Improvement Tracker](https://docs.google.com/spreadsheets/d/1xyqyGUF-oe3x9ln88YonVeOMWWdknik74lVgL_3dBY8/edit#gid=0)
+
+:::
+
+## :mega: Update (Done, Shipped, Made Progress)
+> To be reviewed Async
+
+- ==@raulk==
```
Comment (daviddias) [inline _meeting-notes/2019-12-10.md]: @raulk from [last week's update](https://github.com/ipfs/testground/blob/master/_meeting-notes/2019-12-03.md#dart-next), what is missing is:

- [ ] Error handling guidelines. https://github.com/ipfs/testground/issues/105
- [ ] Start conceptualising multiple TB transfer test case design.
- [ ] Initiate SPEC overhaul to align with actual implementation.

Might share an update and a note if you plan to drop/get done this week?

File: _meeting-notes/2019-12-10.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+- ==@daviddias==
```
Comment (daviddias) [inline _meeting-notes/2019-12-10.md]: @daviddias (me) from @raulk from [last week's update](https://github.com/ipfs/testground/blob/master/_meeting-notes/2019-12-03.md#dart-next), what is missing is:

- [ ] ~~Collect architecture decisions that need to be captured and written down~~ 
  - We are now in the progress of benchmarking Docker Swarm vs. Kubernetes (owner: @nonsense)
  - I missed to inquiry everyone and compile this list. The preliminary list is:
    - We need to document how Sidecar is integrated https://github.com/ipfs/testground/issues/221
    - We need describe where is the line that separates Testground from a Test Plan https://github.com/ipfs/testground/issues/224

File: _meeting-notes/2019-12-10.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+  - Planned Onboarding Hack Week https://docs.google.com/document/d/18P1EfbJ3T4SkafNur7u9uI7WqP8pReP340YXvy8EMAA/edit#
+  - Kicked off the Quarter Retrospective https://github.com/ipfs/testground/issues/220
+  - Unpacked Test Plan 4 for @jimpick https://github.com/ipfs/testground/issues/96#issuecomment-562877707
+  - Reviewed PRs
+  - Unblocked @hacdias
+  - Created issue templates https://github.com/ipfs/testground/pull/229
+  - Groomed the Kanban
+- ==@stebalien==
```
Comment (daviddias) [inline _meeting-notes/2019-12-10.md]: Great update, @Stebalien. Thank you listing what you've achieved and what you haven't.

File: _meeting-notes/2019-12-10.md (hunk 1 above)
Comment (daviddias) [reply _meeting-notes/2019-12-10.md]: Oh I see, this line is a c&p from last week and your update is below.

File: _meeting-notes/2019-12-10.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+        - Working on further "unpacking" - a new "docs" directly under the testplan with a complete "brain-dump" / plan - work-in-progress, will ask for review on PR tomorrow https://github.com/ipfs/testground/pull/232
+    - Test SideCar and demonstrate how to use it (docs or recorded demo)
+        - I reviewed the Sidecar code, but didn't write a new test yet that uses it
+    - Tested Anton's Infra PR
+- ==@nonsense==
+    - Got the `daemon` PR merged
+    - Got the initial `infra` PR (`terraform` and `ansible` playbooks) reviewed and merged
+    - Took over infra from Jim
+    - Added redirect from http://testground.ipfs.team to github.com/ipfs/testground :) testground.ipfs.team is now managed within the Testground team
+- ==@hacdias==
```
Comment (daviddias) [inline _meeting-notes/2019-12-10.md]: @hacdias thank you for sharing a good update!

File: _meeting-notes/2019-12-10.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+          - Solution: [TODO] Stagger the bootstrap step so peers bootstrap into an existing DHT instead of a fresh DHT.
+- ==@jimpick==
+    - Test Plan 4
+        - Discussed ideas with David
+        - David "unpacked" it https://github.com/ipfs/testground/issues/96#issuecomment-562877707
+        - Working on further "unpacking" - a new "docs" directly under the testplan with a complete "brain-dump" / plan - work-in-progress, will ask for review on PR tomorrow https://github.com/ipfs/testground/pull/232
+    - Test SideCar and demonstrate how to use it (docs or recorded demo)
+        - I reviewed the Sidecar code, but didn't write a new test yet that uses it
+    - Tested Anton's Infra PR
+- ==@nonsense==
```
Comment (daviddias) [inline _meeting-notes/2019-12-10.md]: @nonsense thank you for sharing a good update!

File: _meeting-notes/2019-12-10.md [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+  - Measure the memory footprint of a go-libp2p node that runs 1000 find peer queries on the mainnet
+  - Use that measure to calculate how many nodes can be run in each AWS instance
+  - Calculate the right number of machines needed for a cluster to run 10000 libp2p nodes
+  - Enhance the Testground experience by removing the overhead of running a plan (devs should not have to worry about infra). Running a plan should be done in one command
+      - @nonsense - this assumes we run a service on AWS and solve issues like:
+          - https://github.com/ipfs/testground/issues/226
+  - Get filebeat to only pump logs at the end of a run (so that it doesn't influence the performance of the test)
+      - @nonsense - this seems to be the lowest priority for the week, right?
+- ==@hacdias==
```
Comment (daviddias) [inline _meeting-notes/2019-12-10.md]: @hacdias can you also take a look at this list at https://github.com/ipfs/testground/pull/233/files#r356468557 and pick some of those to get done this week?

File: _meeting-notes/2019-12-10.md (hunk 4 above)
Comment (hacdias) [reply _meeting-notes/2019-12-10.md]: So, I stole some: #191 #124 and #158 :)

File: _meeting-notes/2019-12-10.md (hunk 10 above)
Comment (daviddias) [reply _meeting-notes/2019-12-10.md]: From https://github.com/ipfs/testground/pull/233/files#r356514633, @hacdias is now owning
- #191
- #124 
- #158

File: _meeting-notes/2019-12-10.md (hunk 4 above)
Comment (hacdias) [reply _meeting-notes/2019-12-10.md]: I'm gonna steal #34 too.

Comment (daviddias) [reply _meeting-notes/2019-12-10.md]: @hacdias is on 🔥🔥🔥🔥

@jimpick come before it is too late and @hacdias shipped them all 😁

Comment (jimpick) [reply _meeting-notes/2019-12-10.md]: Hey, I was sleeping! I guess I get what is left over. :-)

Thoughts on the 5 remaining ones:

#173 Recreate-broken containers

I'll try to reproduce the failing scenario and do a quick fix.

#122 RunEnv.initLogger is racy

I'll take a look.

#54 Test the TestGround

Moving target, big scope. I labelled it as an "Epic". Needs to be broken down into smaller tasks that can be completed in a short time frame.

#118 Use go.mod for the build step

I think this is missing a design blueprint.

#131 Make Runner.Run() and Builder.Build() take contexts

I can try to add these.

Comment (daviddias) [reply _meeting-notes/2019-12-10.md]: Ok, updating your assignments list

**Pull request merged**